From bfdea14615aa11452c141eadf1102b15a7a10575 Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Thu, 2 May 2013 15:55:15 +0800
Subject: [PORT FROM MAIN] libstagefright: remove 10ms polling in AwesomePlayer

BZ: 34320

AwesomePlayer will wake up every 10ms while video is playing.

This patch will remove the 10ms polling.

And it will save 18mW according to Melwyn's measurement data.

Change-Id: I0ee1079fb3565bc94cc1bee4ed4cae5d5b527a8a
Orig-Change-Id: I9eb1b44f27f8f7338090dff68f1b465ea160e551
Signed-off-by: Richard Tang <richard.tang@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/AwesomePlayer.cpp |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index f3c0908..90dc5c3 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -1842,9 +1842,13 @@ void AwesomePlayer::onVideoEvent() {
             }
         }
 
-        if (latenessUs < -10000) {
+        if (latenessUs < -30000) {
+            // We're more than 30ms early.
+            postVideoEvent_l(30000);
+            return;
+        } else if (latenessUs < -10000) {
             // We're more than 10ms early.
-            postVideoEvent_l(10000);
+            postVideoEvent_l(-latenessUs);
             return;
         }
     }
-- 
1.7.4.1

